app/views/shared/_nav.html.erb

```diff
@@ -4,12 +4,35 @@
       <%= link_to 'Erudition', root_path, class: 'navbar-brand' %>
     </div>
     <div class="navbar-header navbar-right">
-      <% if !user_signed_in? %>
-        <%= link_to 'Log in', new_user_session_path, class: 'btn btn-default navbar-btn nav-pills' %>
-      <% end %>
       <% if user_signed_in? %>
-        <p class="navbar-user hidden-xs"><%= current_user.email %></p>
-        <%= link_to 'Log out', destroy_user_session_path, method: :delete, class: 'btn btn-default navbar-btn' %>
+        <div class="btn-group">
+          <button type="button" class="btn btn-default dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+            <%= current_user.email %> <span class="caret"></span>
+          </button>
+          <ul class="dropdown-menu dropdown-menu-right">
+            <li>
+              <%= link_to 'Your Profile', user_path(current_user.id) %>
+            </li>
+            <li>
+              <%= link_to 'Dashboard', '#' %>
+            </li>
+            <% if current_user.is_admin? %>
+              <li role="separator" class="divider"></li>
+              <li>
+                <%= link_to 'Admin Dashboard', decks_path %>
+              </li>
+            <% end %>
+            <li role="separator" class="divider"></li>
+            <li>
+              <%= link_to 'Edit Profile', edit_user_registration_path(current_user.id) %>
```

Comment:
This is the link to edit the user registration, which i _think_ i used for stuff like password reset. You'll want to use `edit_user_path` instead, and then make a new file `app/views/users/edit.html.erb` to put the form that you have written in `app/views/devise/registrations/edit.html.erb`. Whether or not you'll want to keep the changes in `app/views/devise/registrations/edit.html.erb` I'm not sure at the moment, you'd want to confirm where that file is used